src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverterSettings.cs

```diff
@@ -0,0 +1,250 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+
+namespace System.Text.Json.Serialization
+{
+    public class JsonConverterSettings
+    {
+        private volatile JsonMemberBasedClassMaterializer _classMaterializerStrategy;
```

Comment:
Why `volatile`?

Reply:
It is created lazily and can be accessed by more than one thread